### PR TITLE
Add Tic-Tac-Toe

### PR DESCRIPTION
**Pull request by TeeSeal**
Please review, senpais.

### CODE REVIEW

File: commands/games/tictactoe.js [hunk 1]
```diff
@@ -0,0 +1,145 @@
+const { Command } = require('discord.js-commando');
+const stripIndents = require('common-tags').stripIndents;
+
+const Currency = require('../../currency/Currency');
+
+const games = [];
+const combinations = [[0, 1, 2], [3, 4, 5], [6, 7, 8], [0, 3, 6], [1, 4, 7], [2, 5, 8], [0, 4, 8], [2, 4, 6]];
+
+module.exports = class BlackjackCommand extends Command {
```
Comment (iCrawl) [inline commands/games/tictactoe.js]: Nice copy-paste.

File: commands/games/tictactoe.js [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+	constructor(client) {
+		super(client, {
+			name: 'blackjack',
```
Comment (iCrawl) [inline commands/games/tictactoe.js]: Ditto.

File: commands/games/tictactoe.js [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+			group: 'games',
+			memberName: 'blackjack',
```
Comment (iCrawl) [inline commands/games/tictactoe.js]: Ditto.

File: commands/games/tictactoe.js [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+			description: 'Challenge someone to a game of tic-tac-toe for donuts!',
+			details: 'Challenge someone to a game of tic-tac-toe for donuts.',
```
Comment (iCrawl) [inline commands/games/tictactoe.js]: If details is the same as description theres no need for it.

File: commands/games/tictactoe.js [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+			guildOnly: true,
+			throttling: {
+				usages: 1,
+				duration: 30
+			},
+
+			args: [
+				{
+					key: 'member',
+					prompt: 'Which user would you like to challenge?',
```
Comment (iCrawl) [inline commands/games/tictactoe.js]: Lowercase first letter, `\n` at the end.

File: commands/games/tictactoe.js [hunk 6]
```diff
@@ -0,0 +1,145 @@
...
+					type: 'member'
+				},
+				{
+					key: 'bet',
+					prompt: 'How many donuts do you want to bet?',
+					type: 'integer',
+					max: 1000,
```
Comment (iCrawl) [inline commands/games/tictactoe.js]: Remove the maximum since you already validate below that anyway.

File: commands/games/tictactoe.js [hunk 7]
```diff
@@ -0,0 +1,145 @@
...
+
+		if (userBalance < bet) {
+			return stripIndents`
+				${user} doesn't have enough 🍩s. Their current account balance is ${userBalance} 🍩s.
+				Please try again later.
+				`;
+		}
+
+		if (games.indexOf(msg.guild.id) > -1) {
+			return msg.reply(`you can't start 2 games of Tic-Tac-Toe one the same server at once.`);
```
Comment (iCrawl) [inline commands/games/tictactoe.js]: *on the same server twice.

File: commands/games/tictactoe.js [hunk 8]
```diff
@@ -0,0 +1,145 @@
...
+					});
+
+					if (responses.size === 0) {
+						const loser = players[turn];
+						const winner = players[turn === 'x' ? 'o' : 'x'];
+
+						Currency.removeBalance(loser.id, bet);
+						Currency.addBalance(winner.id, bet);
+
+						return msg.say(`${loser.displayName} failed to respond in time. ${winner.displayName} won their ${bet} 🍩s!;`);
```
Comment (iCrawl) [inline commands/games/tictactoe.js]: Prolly wanna go with *wins their bet.

File: commands/games/tictactoe.js [hunk 9]
```diff
@@ -0,0 +1,145 @@
...
+						turn = turn === 'x' ? 'o' : 'x';
+
+						if (this.gameWon(field)) {
+							const loser = players[turn === 'x' ? 'o' : 'x'];
+							const winner = players[turn];
+
+							Currency.removeBalance(loser.id, bet);
+							Currency.addBalance(winner.id, bet);
+
+							return msg.say(`Game over! ${winner.displayName} won ${bet} 🍩s from ${loser.displayName}.`);
```
Comment (iCrawl) [inline commands/games/tictactoe.js]: Can omit the "from ${loser.displayName}."

File: commands/games/tictactoe.js [hunk 10]
```diff
@@ -0,0 +1,145 @@
...
+			${field[3]} | ${field[4]} | ${field[5]}    |   Turn: ${turn} - ${players[turn].displayName}
+			-----------   |   Type the number of the space you want to occupy.
+			${field[6]} | ${field[7]} | ${field[8]}    |
+			\`\`\`
+			`;
+	}
+
+	gameWon(field) {
+		for (let combo in combinations) {
+			if (field[combo[0]] === field[combo[1]] && field[combo[1]] === field[combo[2]]) {
```
Comment (iCrawl) [inline commands/games/tictactoe.js]: I guess you can inline the return here.

File: commands/games/tictactoe.js (hunk 7 above)
Comment (TeeSeal) [inline commands/games/tictactoe.js]: I actually limited it to 1 game per server for now so this spelling is fine. (except for on*)
"You can't start a game of Tic-Tac-Toe on the same server twice" would imply multiple games on the same server being a possibility for multiple users.

Comment (iCrawl) [inline commands/games/tictactoe.js]: Oh yea, I just made a mistake writing that sentence out thats all. I obv meant the *on

File: commands/games/tictactoe.js [hunk 11]
```diff
@@ -0,0 +1,140 @@
...
+					prompt: 'how many donuts do you want to bet?\n',
+					type: 'integer',
+					validate: async (msg, bet) => {
+						bet = parseInt(bet);
+						const balance = await Currency.getBalance(msg.author.id);
+
+						if (balance < bet) {
+							return stripIndents`
+								You don't have enough donuts. Your current account balance is ${balance} 🍩s.
+								Please specify a valid amount of donuts.
```
Comment (alucard0134) [inline commands/games/tictactoe.js]: The amount is valid, they are just too fucking poor

**Pull request closed without merging**